### PR TITLE
Add remove to-do

### PR DESCRIPTION
**Pull request by BranBayou**
- I created a new branch add-remove for this project.
- I implement a function for adding a new task.
- I implement a function for deleting a task.
- I implement a function for editing task descriptions.
- I stored all the data in local storage.
- I checked for linter errors.

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -1,36 +1,103 @@
...
+  } else {
+    const todo = {
+      description: todoValue,
+      completed: false,
+      index: todos.length,
+    };
+    todos.push(todo);
+    todoInput.value = '';
+    toLocalStorage();
+    renderTodos();
```
Comment (AmaduKamara) [inline src/index.js]: - [ ] In the requirements of this project, you're required to have the index of the todos represent the length of the array. That means, if the array is having 5 items, you should make all the 5th item in the array has an index of 5 and not 4th(not the normal way array counts). You will achieve this with this syntax `index: todos.length + 1,` 
Currently, the todos index starts counting at 0. Kindly use the format above. 👍🏽

Screenshot 2023-04-08 at 5 17 47 PM

File: src/index.js [hunk 2]
```diff
@@ -1,36 +1,103 @@
...
+      editIcon[i].addEventListener('click', () => {
+        handleEdit(i);
+      });
+    }
+    // Function to handle delete action
+    const handleDelete = (index) => {
+      todos.splice(index, 1);
+      toLocalStorage();
+      renderTodos();
+    };
```
Comment (AmaduKamara) [inline src/index.js]: - [ ] When a todo is deleted, you are to update the indexes of the remaining todos to be in order as they are. 
In the screenshot below, I entered 6 todos, and I deleted them randomly, but the index is not organized. You're required to organize the order. In short, the indexes need to update to be in order 1...2...3.

Screenshot 2023-04-08 at 5 26 27 PM

**Review by AmaduKamara — CHANGES_REQUESTED** [review]
# Status: Required Changes ♻️ ♻️ ♻️

Hi @BranBayou,

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
✅ You used Gitflow correctly 👍🏽
✅ Linter checks passed 👍🏽
✅ Good descriptive PR 👍🏽
✅ Add, delete, and update features work fine 👍🏽

## Required Changes ♻️
_Check the comments under the review._

- [ ] Kindly refer to the project requirements on the screenshot below and check the inline comments to see the updates you need to work on 👍🏽

Screenshot 2023-04-08 at 5 31 11 PM

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

**Review by GedeonTS — APPROVED** [review]
# STATUS: APPROVED :tada: :clap: :champagne: 
Hi @BranBayou ,

You are amazing :smiley: :clap: !  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

 

## Highlights :clap: :tophat: 
- No linter errors :+1: 
- Good use of GitHub flow :+1: 
- ADD, REMOVE and UPDATE functionalities are perfectly working :clap: 
## Optional suggestions :small_airplane: 

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**